src/metatrain/utils/metadata.py

```diff
@@ -17,6 +17,8 @@ def append_metadata_references(self: ModelMetadata, other: ModelMetadata) -> Non
         if key not in self_dict["references"]:
             self_dict["references"][key] = values
         else:
-            self_dict["references"][key] += values
```

Comment:
Should we turn this function into a more general `merge_metadata(existing, new)`? So more than just the references can be merged when fine-tuning